loader/loader_windows.c

```diff
@@ -996,4 +998,60 @@ VkResult windows_sort_physical_device_groups(struct loader_instance *inst, const
     return VK_SUCCESS;
 }
 
+char *windows_get_app_package_manifest_path(const struct loader_instance *inst)
+{
+    UINT32 numPackages = 0, bufferLength = 0;
+    PCWSTR familyName = L"Microsoft.D3DMappingLayers_8wekyb3d8bbwe";
+    if (ERROR_INSUFFICIENT_BUFFER != GetPackagesByPackageFamily(familyName,
+                                                                &numPackages, NULL,
+                                                                &bufferLength, NULL) ||
+        numPackages == 0 || bufferLength == 0) {
+        loader_log(inst, VULKAN_LOADER_INFO_BIT, 0,
+                   "windows_get_app_package_manifest_path: Failed to find mapping layers packages by family name\n");
+        return NULL;
+    }
+
+    char *ret = NULL;
+    WCHAR *buffer = loader_instance_heap_alloc(inst, sizeof(WCHAR) * bufferLength, VK_SYSTEM_ALLOCATION_SCOPE_COMMAND);
+    PWSTR *packages = loader_instance_heap_alloc(inst, sizeof(PWSTR) * numPackages, VK_SYSTEM_ALLOCATION_SCOPE_COMMAND);
+    if (!buffer || !packages) {
+        loader_log(inst, VULKAN_LOADER_ERROR_BIT, 0,
+                   "windows_get_app_package_manifest_path: Failed to allocate memory for package names\n");
+        goto cleanup;
+    }
+
+    if (ERROR_SUCCESS != GetPackagesByPackageFamily(familyName,
+                                                    &numPackages, packages,
+                                                    &bufferLength, buffer)) {
+        loader_log(inst, VULKAN_LOADER_ERROR_BIT, 0,
+                   "windows_get_app_package_manifest_path: Failed to mapping layers package full names\n");
+        goto cleanup;
+    }
+
+    UINT32 pathLength = 0;
+    WCHAR path[MAX_PATH];
+    if (ERROR_INSUFFICIENT_BUFFER != GetPackagePathByFullName(packages[0], &pathLength, NULL) ||
+        pathLength > MAX_PATH ||
+        ERROR_SUCCESS != GetPackagePathByFullName(packages[0], &pathLength, path)) {
+        loader_log(inst, VULKAN_LOADER_ERROR_BIT, 0,
+                   "windows_get_app_package_manifest_path: Failed to get mapping layers package path\n");
+        goto cleanup;
+    }
+
+    int narrowPathLength = WideCharToMultiByte(CP_ACP, 0, path, -1, NULL, 0, NULL, NULL);
+    ret = loader_instance_heap_alloc(inst, narrowPathLength, VK_SYSTEM_ALLOCATION_SCOPE_COMMAND);
+    if (!ret) {
+        loader_log(inst, VULKAN_LOADER_ERROR_BIT, 0,
+                   "windows_get_app_package_manifest_path: Failed to allocate path\n");
+        goto cleanup;
+    }
+
+    WideCharToMultiByte(CP_ACP, 0, path, -1, ret, narrowPathLength, NULL, NULL);
```

Comment:
Is `WideCharToMultiByte` always going to work? I'd be more confident if a potential error was at least reported, even if not acted upon.

Reply:
Fair, I'll add some error handling.